listenbrainz/webserver/views/api_compat_deprecated.py

```diff
@@ -156,8 +156,14 @@ def _to_native_api(data, append_key):
     # if this is not a now playing request, get the timestamp
     if append_key != '':
         try:
-            listen['listened_at'] = data['i{}'.format(append_key)]
-        except KeyError:
+            listen['listened_at'] = int(data['i{}'.format(append_key)])
+        except (KeyError, ValueError):
+            return None
+
+        # if timestamp is too high, this is an invalid listen
+        # in order to make up for possible clock skew, we allow
+        # timestamps to be one hour ahead of server time
+        if listen['listened_at'] > int(time()) + 86400:
```

Comment:
Now we have two timestamp upper limit checkers, with different values. Why is this one 24 hours? I think we should have one function for this.

Reply:
whoops.